drivers/modem/modem_iface_uart.c

```diff
@@ -228,3 +229,27 @@ int modem_iface_uart_init(struct modem_iface *iface,
 
 	return 0;
 }
+
+void modem_iface_uart_sleep(struct modem_iface *iface, uint32_t state)
+{
+	uart_irq_rx_disable(iface->dev);
```

Comment:
Though valuable, this saves very little power. I don't think we have a mechanism to Wake / Sleep the actual modem yet, which will save tons of power.. Any plans on adding that support ?

Reply:
You could look to implement AT+CFUN command; switch between 0 & 1 or maybe using 4.

> AT+CFUN AT command  sets the level of functionality in the MT. Level "full functionality" is where the highest level of power is drawn. "Minimum functionality" is where minimum power is drawn.
> Possible values are,
> 0 minimum functionality
> 1 full functionality
> 2 disable phone transmit RF circuits only
> 3 disable phone receive RF circuits only
> 4 disable phone both transmit and receive RF circuits

Reply:
Yeah, that's a Modem specific detail. One can even just toggle the MDM_EN pin.

I'm more interested in a proper infrastructure within Zephyr to support this feature, which is not available, as far as I know..

Reply:
We do have a patch to add a power saving mechanism for the sara-r4 modem using device_pm.
This means that the device should be put in active mode before creating and using a socket and should be put back in sleep after all socket are closed.
However this solution is more of a hack and is therefore not clean enough to be upstreamed.

Reply:
Commit removed.